### DISCUSSION TITLE
[DONE ✅]  [UX/UI] Drop the page header elements to the right column

### DISCUSSION THREAD

**Discussion by dstrukt** [btcpayserver/btcpayserver#2362] [General]
We currently have a pattern where we place the header above the page sections / main content. 

I would like to propose a simple change of moving the header over to the right column on all of these instances, removing the green divider element (as our typography should handle hierarchy just fine here), and removing the "Manage store: " pre-header text (on the pages where it appears e.g. like "Manage your account: Profile" on the profile screens), opting for a more simple title (General, Rates, etc...).

(Still debating the breadcrumbs .. trying to see if there's a better navigational pattern before I advocate for support)

Before:
<img width="1830" alt="Screen Shot 2021-03-14 at 2 16 36 PM" src="https://user-images.githubusercontent.com/6250771/111084711-fe555000-84d0-11eb-8a1a-8311930de8f4.png">

After:
<img width="1149" alt="Screen Shot 2021-03-14 at 2 24 49 PM" src="https://user-images.githubusercontent.com/6250771/111084732-188f2e00-84d1-11eb-8f64-e6be0d9187a2.png">

Open to suggestions / feedback.

**Comment by pavlenex** [comment]
Concept ACK. What would, in general, be an alternative solution to breadcrumbs? I personally like breadcrumbs but unsure how others feel about it.

**Reply by Kukks** [reply]
I think this looks great. The breadcrumbs are way more descriptive than our generic "Main Title". For example, in wallets, I  often find myself having to go back to the list and entering again to double-check I'm in the right wallet.

**Comment by Kukks** [comment]
Concept ACK

**Comment by bolatovumar** [comment]
Looks good but now I'm wondering if this header is needed at all? You click on the "General" tab and see the heading which just says "General" which is just the same thing you clicked on. I would assume it would be the same for other tabs? In this case it feels like you could just omit the heading entirely, no?

**Reply by pavlenex** [reply]
Correct, I also believe removing the heading entirely makes sense if breadcrumbs are added.

**Reply by dennisreimann** [reply]
Yup, will build this on top of #2378.

**Reply by ghost** [reply]
concept ACK for new design without "General" title twice. 
always good to remove UI clutter where the same thing is mentioned twice.

**Reply by dstrukt** [reply]
I think I initially I added the large page header because I believe there were instances where we wouldn't/won't have breadcrumbs (might have added it in the design pre-breadcrumbs too, can't remember), nor left side navigation.

However, I suppose we could simply add the title in instances where that might be needed. 

I think I was also thinking it would enable us to give the page a different header, not exactly the same as the left navigation, but I'm all for stripping back any necessary labeling / UI, so if it makes sense, let's do it!

(just sharing my 2 sats incase for more context)

**Reply by dennisreimann** [reply]
My preference would also be to not duplicate the title (in addition to having it in the breadcrumb) – nevertheless I think it needs some kind of accentuation. 

Here's the plain version: 

![breadcrumb](https://user-images.githubusercontent.com/886/112027613-a0131780-8b37-11eb-9d1f-bfc39182da13.png)

Here's a version with the active item highlighted:

![Bildschirmfoto 2021-03-22 um 17 55 47](https://user-images.githubusercontent.com/886/112028193-36473d80-8b38-11eb-9804-7d616dd83f96.png)

Open for discussion :)

**Reply by ghost** [reply]
ACK second version with highlights.

The new design is coming together nicely, well done!

**Reply by pavlenex** [reply]
@dennisreimann can you elaborate on what you mean by `accentuation`? It feels cleaner but in a way too empty and we should probably think of ways to better present it. I def prefer this to what we have. The second option seems to be in line with what we have in #2378

**Reply by dennisreimann** [reply]
Now that the headline has been merged into the last/active breadcrumb item, we need to somehow highlight it a bit – that's what I meant. Maybe @dstrukt has some input here :)

**Reply by dstrukt** [reply]
This is the kind of instance I was referring to with the header/breadcrumbs in previous posts, but I'm glad to see we're already running up against the issues, and think it'd be worth talking through some of this at our design meeting as well.

1. I agree it does feel a bit empty without the header...and it's possible a header could be warranted in some cases.

2. In regards to the breadcrumbs, with the new updated "active" header elements, they feel a bit redundant, and I had planned to make a more detailed post to outline the fairly restrictive rules where they would appear (why I didn't include the breadcrumbs as part of this / still vetting if they're a feasible solution). Perhaps I should have made this more clear. 

This *may* change your minds about this implementation, but that's why I didn't push for it right away :) and what I've been wrestling with to see if it will work ... it would ultimately be nice to not need them at all, but we're still a little ways away from that.

For example, I don't think we should show the breadcrumbs on instances where the navigation isn't 2-3 layers deep or not immediately accessible / clear where the user is, like:

- Profile
<img width="1868" alt="Screen Shot 2021-03-22 at 12 55 19 PM" src="https://user-images.githubusercontent.com/6250771/112051477-367f1300-8b0f-11eb-98c1-4ef2a51ed535.png">
- Server Settings
<img width="1621" alt="Screen Shot 2021-03-22 at 12 55 32 PM" src="https://user-images.githubusercontent.com/6250771/112051482-37b04000-8b0f-11eb-96de-accbb1c26ed6.png">

The active header, and the left navigation is sufficient for all levels of information, and it's very clear where you are at all times, thus no need foe breadcrumbs.

 
Alternatively, we should have them on:

- The "create" screens, if we keep the separate page flow, vs an overlay or popup
<img width="1168" alt="Screen Shot 2021-03-22 at 12 55 49 PM" src="https://user-images.githubusercontent.com/6250771/112051567-5282b480-8b0f-11eb-8a61-e934992d7fdc.png">
<img width="1164" alt="Screen Shot 2021-03-22 at 1 21 19 PM" src="https://user-images.githubusercontent.com/6250771/112053392-8a8af700-8b11-11eb-978e-89af0b852a4e.png">
<img width="820" alt="Screen Shot 2021-03-22 at 1 26 30 PM" src="https://user-images.githubusercontent.com/6250771/112053996-45b39000-8b12-11eb-9c10-9f1c685463c5.png">
<img width="1164" alt="Screen Shot 2021-03-22 at 1 21 19 PM" src="https://user-images.githubusercontent.com/6250771/112054034-506e2500-8b12-11eb-84db-b52bb006b00f.png">

- Wallets (to help navigate the odd structure / unclear location as @Kukks pointed out earlier). I'm still not entirely sure if there is potentially a better way, but it's a feasible solution for the time being.
<img width="1019" alt="Screen Shot 2021-03-22 at 1 22 32 PM" src="https://user-images.githubusercontent.com/6250771/112054004-4815ea00-8b12-11eb-88e1-da8c6310c69b.png">

- Stores (to help navigate the odd structure)
<img width="1773" alt="Screen Shot 2021-03-22 at 1 27 33 PM" src="https://user-images.githubusercontent.com/6250771/112054099-64198b80-8b12-11eb-95d4-61c7f992fc18.png">

- Viewing an invoice (but this could be handled as an aside/overlay in the future like create screens)

Honestly, the create screens and the store/wallet are the only places where they really make sense, and part of that is just because the relationship between Wallet/Store is a bit strange at the moment.

If any of what i've said is not clear, I'm happy to provide more context, but hopefully everything is a little more clear looking at this .. panning around in the Figma would highlight this issue quite well too.

**Reply by dennisreimann** [reply]
I tinkered a bit more with this and now think we should solve this without the breadcrumbs. I think it's kind of confusing and inconsistent to sometimes show them while hiding them most of the time.

The general problem exists, because we have different kinds of depths inside the main categories …

## Simple case

Server Settings contains both cases, e.g.:
- Users/Services: A list of separate entities, each with management functionality on their own (on  another level of subpages)
- Email/Policies/Theme: A single page to manage these things

However, for the Server Settings both cases work well without any extras like the breadcrumb, because we can offer enough guidance using the navigation and the page headline.

![server-users](https://user-images.githubusercontent.com/886/112340229-a7196180-8cc0-11eb-9898-6293016817a0.png)

![server-user](https://user-images.githubusercontent.com/886/112340224-a680cb00-8cc0-11eb-8427-d0d4ed2d9b41.png)

![server-email](https://user-images.githubusercontent.com/886/112340219-a5e83480-8cc0-11eb-83b6-bb559e9e884d.png)

## Nested case

Stores, Wallets, etc. need a bit more guidance though, because when you drill down you lose context without the name of the entity. This case can be solved with an additional headline (substituting the breadcrumb): This way we keep the design consistent and the user still has a good way of navigating back to the list via the main menu. The user can drill down into the different functionalities of the entity via the side menu, which also offers context …

### Stores

![stores](https://user-images.githubusercontent.com/886/112341015-5bb38300-8cc1-11eb-9d6e-28f009bddd9d.png)

For the individual entity, the headline in this case shows the store name …

![store](https://user-images.githubusercontent.com/886/112341052-61a96400-8cc1-11eb-8535-bde18711d691.png)

For the Create case I think it's a good addition to have the "Back to list" beside the save button, so that you can also easily go back/cancel when scrolled down (e.g. when you are on a longer page like create payment request). 

![store-create](https://user-images.githubusercontent.com/886/112341073-64a45480-8cc1-11eb-941a-2be6530a1ea4.png)

### Wallets

Same works well for the wallets.

![wallets](https://user-images.githubusercontent.com/886/112343169-2740c680-8cc3-11eb-895c-13c25bf2aecc.png)

![wallet](https://user-images.githubusercontent.com/886/112343177-2a3bb700-8cc3-11eb-84b5-7216836c0c0f.png)

![wallet-send](https://user-images.githubusercontent.com/886/112343202-2f990180-8cc3-11eb-88d4-ba0e7c13a813.png)


I think this is clean, consistent and intuitive :)

Another idea we might want to think about: Making the left menu sticky, so that it is always visible, even when scrolling down longer pages.

**Reply by dstrukt** [reply]
I like where this is going!

- ACK the left sticky menu, how I had always hoped it would work -- will explore having keep the old header (like you have here instead of pushing it over to the right -- few things I want to experiment with) I also have a mobile UI/UX proposal for the left sidebar, but need to explore it a bit further.
-- NOTED FOR DESIGN REVIEW

- I have improvement suggestions for Wallets/Stores/etc.. listing pages that I think will help address some of the issues you raised, but planned to make a bigger ticket about this as there's still some considerations left .. impressed we're already at a stage where we could potentially talk to this.
-- NOTED FOR DESIGN REVIEW

- I'm not a big fan of the "Back to List" UI, in that one instance I prefer the breadcrumbs, but there are a few potential ways to address create flows that could be explored, so I'm happy to roll with this for the time being. We could address the options in the design meeting, I have a few of the concepts mocked, and it would be great to get feedback so I can standardize that pattern for future screens.
![image](https://user-images.githubusercontent.com/6250771/112378939-c6a29100-8ca4-11eb-88f7-0022ab914b6f.png)
-- NOTED FOR DESIGN REVIEW

- It's a small nitpick, and I like the button action in the top/right (a pattern I carry through the Figma as well), but I think we'll need to make it consistent/dependable vs. semi-floating like it is now.
-- NOTED FOR DESIGN REVIEW

- Just a note, not necessarily a change to make now, and it ties into the default state ticket I made earlier, leaving the CTA and text and hiding up empty tables and only showing when there is data, e.g.
![image](https://user-images.githubusercontent.com/6250771/112379373-4f213180-8ca5-11eb-9a7f-96c9a2422b2c.png)

Some of this is a bit tangential to things, but it's all moving in the right direction, and I wanted to get my thoughts out there, even if this is all expressed in 4-5 different PRs.

The future bootstrap changes are going to help a lot with this as well.

**Comment by dennisreimann** [comment]
I've now prepared everything according to what we discussed in the [design meeting](https://github.com/btcpayserver/organization/issues/29) (shown in [this comment](https://github.com/btcpayserver/btcpayserver/discussions/2362#discussioncomment-525527)).

As this affects every major view of the app, I recommend the following procedure:
1. Wait for security release
2. Merge open UI-related PRs (#2378 which my branch is based on, but also #2393, #2373, #2348, #2305)
3. Rebase branch and update views affected by the merged PRs
4. Send PR for this

**Reply by pavlenex** [reply]
Good plan, Dennis, agreed with the outlined steps, excited! 🚀

**Comment by pavlenex** [comment]
@dstrukt @dennisreimann Did we find a consensus on this one or has this been added in satisfactory way?

**Reply by dstrukt** [reply]
Indeed we did.

**Reply by dennisreimann** [reply]
Yup, done 🚀
